gui/src/app/StanSampler/StanSampler.ts

```diff
@@ -12,6 +12,17 @@ export type SamplingOpts = {
     seed: number | undefined
 }
 
+export const isSamplingOpts = (x: any): x is SamplingOpts => {
+    if (!x) return false
+    if (typeof x !== 'object') return false
+    if (typeof x.num_chains !== 'number') return false
+    if (typeof x.num_warmup !== 'number') return false
+    if (typeof x.num_samples !== 'number') return false
+    if (typeof x.init_radius !== 'number') return false
+    if (x.seed !== undefined && typeof x.seed !== 'number') return false
+    return true
+}
```

Comment:
Should we also verify that e.g. `num_chains` is an integer? 

Not sure what the best way to check this is in JS -- `Math.round(x) === x`?

Reply:
We could check that elsewhere, but I don't think this belongs in the type guard. I believe a type guard should have the sole purpose of guaranteeing that the thing if of the right javascript type.